pandas/io/parsers.py

```diff
@@ -1896,23 +1901,46 @@ def __init__(self, src, **kwds):
 
         ParserBase.__init__(self, kwds)
 
-        encoding = kwds.get("encoding")
-
-        # parsers.TextReader doesn't support compression dicts
-        if isinstance(kwds.get("compression"), dict):
-            kwds["compression"] = kwds["compression"]["method"]
+        if kwds.get("memory_map", False):
+            # memory-mapped files are directly handled by the TextReader.
+            # compression and file-object were never supported by it.
+            src = stringify_path(src)
 
-        if kwds.get("compression") is None and encoding:
-            if isinstance(src, str):
-                src = open(src, "rb")
-                self.handles.append(src)
+            if not isinstance(src, str):
+                raise ValueError(
+                    "read_csv does support only string-like objects with "
+                    + "memory_map=True and engine='c'. Falling back to "
+                    + "memory_map=False. Please use engine='python' instead."
+                )
```

Comment:
If it's an error, then we should change the error message.

Reply:
ups, yes :)

Reply:
done